### PR TITLE
Create database entities and repository class for per user search history

### PR DESCRIPTION
**Pull request by bowenduanmu**
Create a new db object and collection to collect data for user search history and create a function to fetch data when user uses google search service.

### CODE REVIEW

**Review by ujcookcao — DISMISSED** [review]
The code works as expected. It LGTM

File: src/main/java/edu/ucsb/mapache/services/GoogleSearchService.java [hunk 1]
```diff
@@ -161,6 +170,30 @@ public void saveToSearchRepository(String searchQuery) {
...
+	    String userid=firstName.concat(lastName);
+        s.setSearchTerm(searchQuery);
+	
+        s.setUserID(userid);
+	    String timestamp = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS Z").format(Calendar.getInstance().getTime());
+        s.setTimestamp(timestamp);
+
+	    usersearchRepository.save(s);
+
+    }
```
Comment (pconrad) [inline src/main/java/edu/ucsb/mapache/services/GoogleSearchService.java:193]: I wonder whether you might make the indentation more consistent.

Comment (pconrad) [reply src/main/java/edu/ucsb/mapache/services/GoogleSearchService.java:193]: Code Formatting in VS Code:
On Windows Shift + Alt + F
On Mac Shift + Option + F
On Linux Ctrl + Shift + I

**Review by pconrad — CHANGES_REQUESTED** [review]

File: src/main/java/edu/ucsb/mapache/services/GoogleSearchService.java (hunk 1 above)
Comment (bowenduanmu) [reply src/main/java/edu/ucsb/mapache/services/GoogleSearchService.java]: Change the indentation.

**Review by pconrad — APPROVED** [review]

**Review by YINGLONGWANG — APPROVED** [review]
LGTM

**Pull request merged**